packages/grid/x-data-grid-pro/src/components/GridColumnMenuPinningItem.tsx

```diff
@@ -12,16 +13,43 @@ function GridColumnMenuPinningItem(props: GridColumnMenuItemProps) {
   const { colDef, onClick } = props;
   const apiRef = useGridApiContext();
   const rootProps = useGridRootProps();
+  const theme = useTheme();
 
-  const pinColumn = (side: GridPinnedPosition) => (event: React.MouseEvent<HTMLElement>) => {
-    apiRef.current.pinColumn(colDef.field, side);
-    onClick(event);
-  };
+  const pinColumn = React.useCallback(
+    (side: GridPinnedPosition) => (event: React.MouseEvent<HTMLElement>) => {
+      apiRef.current.pinColumn(colDef.field, side);
+      onClick(event);
+    },
+    [apiRef, colDef.field, onClick],
+  );
 
   const unpinColumn = (event: React.MouseEvent<HTMLElement>) => {
     apiRef.current.unpinColumn(colDef.field);
     onClick(event);
   };
+  const renderLeftSideMenuItem = React.useCallback(
+    () => (
+      <MenuItem onClick={pinColumn(GridPinnedPosition.left)}>
+        <ListItemIcon>
+          <rootProps.components.ColumnMenuPinLeftIcon fontSize="small" />
+        </ListItemIcon>
+        <ListItemText>{apiRef.current.getLocaleText('pinToLeft')}</ListItemText>
+      </MenuItem>
+    ),
+    [apiRef, rootProps, pinColumn],
+  );
+
+  const renderRightSideMenuItem = React.useCallback(
+    () => (
+      <MenuItem onClick={pinColumn(GridPinnedPosition.right)}>
+        <ListItemIcon>
+          <rootProps.components.ColumnMenuPinRightIcon fontSize="small" />
+        </ListItemIcon>
+        <ListItemText>{apiRef.current.getLocaleText('pinToRight')}</ListItemText>
+      </MenuItem>
+    ),
+    [apiRef, rootProps, pinColumn],
+  );
```

Comment:
We can use the variable instead of a callback wrapper here:
```suggestion
  const pinToRightMenuItem = (
    <MenuItem onClick={pinColumn(GridPinnedPosition.right)}>
      <ListItemIcon>
        <rootProps.components.ColumnMenuPinRightIcon fontSize="small" />
      </ListItemIcon>
      <ListItemText>{apiRef.current.getLocaleText('pinToRight')}</ListItemText>
    </MenuItem>
  );
```